src/Net/Http.fs

```diff
@@ -554,20 +554,39 @@ module private HttpHelpers =
         hasContentType.Value
 
     let getResponse (req:HttpWebRequest) silentHttpErrors =
-        if defaultArg silentHttpErrors false then
+
+        let getResponseAsync (req:HttpWebRequest) =
             async {
-                try
+                #if FX_NO_WEBREQUEST_TIMEOUT
                     return! Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
-                with
-                    | :? WebException as exn -> 
-                        if exn.Response <> null then 
-                           return exn.Response
-                        else 
-                            reraisePreserveStackTrace exn
-                            return Unchecked.defaultof<_>
+                #else
+                    let! child = 
+                        Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
+                        |> fun x -> Async.StartChild(x, req.Timeout)
```

Comment:
The "default" default is 100s, but you're right in that we can override it with `Timeout.Infinite` at the time the request is created. For some reason I thought these functions should work with any provided `HttpWebRequest`, but it doesn't seem to be the case and we're creating it internally.

I'll take a look at doing these changes at some point next week.